### PR TITLE
feat(sql): add iso-8601 format support for intervals

### PR DESCRIPTION
**Pull request by etolbakov**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link
https://github.com/GreptimeTeam/greptimedb/issues/4215

## What's changed and what's your intention?

Add support for the format of ISO 8601 for the interval type.

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
  - Added support for ISO 8601 format interval parsing and normalization.

- **Improvements**
  - Enhanced interval transformation by distinguishing between interval abbreviations and ISO 8601 formats.
  - Improved interval string processing capabilities.

- **Tests**
  - Added new test cases demonstrating SQL queries with intervals in ISO 8601 format and interval arithmetic operations.

### CODE REVIEW

File: src/sql/src/statements/transform/expand_interval.rs [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use core::time::Duration as StdDuration;
```
Comment (evenyag) [inline src/sql/src/statements/transform/expand_interval.rs:15]:
```suggestion
use std::time::Duration as StdDuration;
```

**Review by evenyag — APPROVED** [review]

File: src/sql/src/statements/transform/expand_interval.rs (hunk 1 above)
Comment (etolbakov) [reply src/sql/src/statements/transform/expand_interval.rs:15]: Thank you! Will make an adjustment

File: src/sql/src/statements/transform/expand_interval.rs [hunk 2]
```diff
@@ -137,26 +138,62 @@ fn update_existing_interval_with_value(interval: &Interval, value: Box<Expr>) ->
...
+                *INTERVAL_ABBREVIATION_MAPPING
+                    .get(mat_str)
+                    .unwrap_or(&mat_str)
+            })
+            .join(" "),
+    )
+}
+
+fn is_iso8601_format(interval_str: &str) -> bool {
+    interval_str.starts_with('P')
```
Comment (killme2008) [inline src/sql/src/statements/transform/expand_interval.rs:193]: What about using a regex? Looks more clear.

**Review by killme2008 — APPROVED** [review]
Great job!

File: src/sql/src/statements/transform/expand_interval.rs (hunk 2 above)
Comment (etolbakov) [reply src/sql/src/statements/transform/expand_interval.rs:193]: Good point indeed! I was adding conditions one by one (lowercase case, then negative and forgot to review with the fresh eye)

**Pull request merged**